prover_rust/src/main.rs

```diff
@@ -0,0 +1,141 @@
+mod config;
+mod coordinator_client;
+mod geth_client;
+mod key_signer;
+mod prover;
+mod task_cache;
+mod types;
+mod utils_log;
+mod version;
+mod zk_circuits_handler;
+
+use anyhow::{Context, Result};
+use config::Config;
+use coordinator_client::listener::Listener;
+use log;
+use prover::Prover;
+use std::rc::Rc;
+use task_cache::TaskCache;
+use types::TaskWrapper;
+
+struct ClearCacheCoordinatorListener {
+    pub task_cache: Rc<TaskCache>,
+}
+
+impl Listener for ClearCacheCoordinatorListener {
+    fn on_proof_submitted(&self, req: &coordinator_client::types::SubmitProofRequest) {
+        let result = self.task_cache.delete_task(req.task_id.clone());
+        if let Err(e) = result {
+            log::error!("delete task from embed db failed, {}", e.to_string());
+        } else {
+            log::info!(
+                "delete task from embed db successfully, task_id: {}",
+                &req.task_id
+            );
+        }
+    }
+}
+
+struct TaskProcessor<'a> {
```

Comment:
make it a separate file `task_processor.rs`

Reply:
done